bin/tx-prover/src/proxy/mod.rs

```diff
@@ -31,6 +37,9 @@ const LOCALHOST_ADDR: &str = "127.0.0.1";
 // WORKER
 // ================================================================================================
 
+/// Worker
+///
+/// A worker is a backend server that processes requests. It is represented by its address.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Worker {
```

Comment:
I would write the comment as:
```
/// A worker used for processing of requests.
///
/// A worker consists of a backend service (defined by worker address) and a flag indicating
/// whether the worker is currently available to process new requests.
```

Reply:
I'm updating this comment, and including the health check client info.